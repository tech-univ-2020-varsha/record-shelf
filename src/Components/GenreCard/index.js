import React from 'react';
import propTypes from 'prop-types';
import Button from '../Button/index';
import * as styles from './index.module.css';
import guitar from '../../images/001-guitar.png';

const GenreCard = ({ cardName }) => (
  <div className={styles.genreCardContainer}>
    <div className={styles.logo}>
      <img src={guitar} alt="Logo" className={styles.genreIcon} />
    </div>
    <div className={styles.cardBtn}>
      <Button>{cardName}</Button>
    </div>
  </div>
);
export default GenreCard;


GenreCard.propTypes = {
  cardName: propTypes.string.isRequired,
};
