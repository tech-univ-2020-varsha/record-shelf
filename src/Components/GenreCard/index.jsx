import React from 'react';
import propTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Button from '../Button/index';
import * as styles from './index.module.css';
import guitar from '../../images/001-guitar.png';
import amplifier from '../../images/006-amplifier.png';
import electricGuitar from '../../images/012-electric guitar.png';
import singer from '../../images/049-singer.png';

const GenreCard = ({ cardName, genreDetails }) => {
  let logo;
  if (cardName === 'rock') { logo = guitar; }
  if (cardName === 'pop') { logo = amplifier; }
  if (cardName === 'bollywood') { logo = electricGuitar; }
  if (cardName === 'country') { logo = singer; }
  return (
    <div className={styles.genreCardContainer}>
      <div className={styles.logo}>
        <img src={logo} alt="Logo" className={styles.genreIcon} />
      </div>
      <div className={styles.cardBtn}>
        <Link to={{ pathname: '/genres/songs', state: { genreType: cardName, genreDetails } }}>
          <Button>{cardName}</Button>
        </Link>
      </div>
    </div>
  );
};


export default GenreCard;


GenreCard.propTypes = {
  cardName: propTypes.string.isRequired,
  genreDetails: propTypes.arrayOf(propTypes.string).isRequired,
};
