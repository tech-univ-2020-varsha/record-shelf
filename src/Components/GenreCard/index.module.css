.genreIcon{
    max-width: 30%;
    z-index: auto;
}

.genreCardContainer{
    background-color: rgb(215,215,215);
    border-radius: 2em;
}

.cardBtn{
    z-index: 10;
}

.logo{
align-self: flex-start;
}

