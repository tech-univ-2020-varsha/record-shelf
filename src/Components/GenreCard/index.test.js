import { render } from '@testing-library/react';
import React from 'react';
import GenreCard from '.';

describe('the genre card', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<GenreCard />);
    expect(asFragment).toMatchSnapshot();
  });
});
