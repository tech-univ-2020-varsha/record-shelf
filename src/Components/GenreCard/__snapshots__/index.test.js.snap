// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the genre card should render correctly 1`] = `[Function]`;
